v1/topdown/topdown_partial_test.go

```diff
@@ -4059,6 +4059,123 @@ func TestTopDownPartialEval(t *testing.T) {
 			},
 			wantQueries: []string{`"x" = input.x`},
 		},
+
+		{
+			note:  "default function",
+			query: "data.test.p = true",
+			modules: []string{`package test
+					default f(x) := true # return true if x.size is undefined
+					f(x) if {
+						x.size < 100
+					}
+					p if {
+						f(input.x)
+					}
+				`},
+			wantQueries: []string{"data.partial.test.f(input.x)"},
+			wantSupport: []string{
+				`package partial.test
+				
+				default f(__local0__3) = true
+				f(__local1__2) = true if { __local2__2 = __local1__2.size; lt(__local2__2, 100) }`,
+			},
+		},
+		{
+			note:  "default function, result comparison (same as default)",
+			query: "data.test.p = true",
+			modules: []string{`package test
+					default f(x) := true # return true if x.size is undefined
+					f(x) if {
+						x.size < 100
+					}
+					p if {
+						f(input.x) == true
+					}
+				`},
+			wantQueries: []string{"data.partial.test.f(input.x, true)"},
+			wantSupport: []string{
+				`package partial.test
+				
+				default f(__local0__3) = true
+				f(__local1__2) = true if { __local3__2 = __local1__2.size; lt(__local3__2, 100) }`,
+			},
+		},
+		{
+			note:  "default function, result comparison (not same as default)",
```

Comment:
This test fails, as we don't want a support module with a default function declaration.